sdk/typescript/src/utils/move-registry.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { TransactionDataBuilder } from '../transactions/TransactionData.js';
+
+/** The pattern to find an optionally versioned name */
+const NAME_PATTERN = /^([a-z0-9]+(?:-[a-z0-9]+)*)@([a-z0-9]+(?:-[a-z0-9]+)*)(?:\/v(\d+))?$/;
+const NAME_SEPARATOR = '@';
+
+export type NameResolutionRequest = {
+	id: number;
+	type: 'package' | 'moveType';
+	name: string;
+};
+
+export type NamedPackagesPluginCache = {
+	packages: Record<string, string>;
+	types: Record<string, string>;
+};
+
+/**
+ * Looks up all `.move` names in a transaction block.
+ * Returns a list of all the names found.
+ */
+export const findTransactionBlockNames = (
```

Comment:
can we move this function into the plugin file.  This isn't at all followed across the SDK, but I prefer not to use imports from other parts of the SDK that aren't publicly exported.  This created a weird cycle of  /transaction/NamedPackagesPlugin -> findTransactionBlockNames ->  /utils/ /transaction/TransactionDataBuilder

I think just leaving all this in the plugin file will be better

Reply:
same for the replaceNames, listToRequests

Reply:
You can create a utils file inside the plugins directory if you want